### ISSUE TITLE
Customize ExpansionPanelList

### ISSUE THREAD

**Issue by sanisloandras** [flutter/flutter#60438]
Hello!

My goal is:
![Screenshot_20200627-211252](https://user-images.githubusercontent.com/29690177/85929112-80a1d380-b8b2-11ea-8415-20d312d774de.jpg)

My current implementation looks like this:
![1](https://user-images.githubusercontent.com/29690177/85929141-a5964680-b8b2-11ea-9eae-95af1fdf4b89.PNG)

Code:
```
ExpansionPanelList buildExpansionPanelList() {
    return ExpansionPanelList(
      expandedHeaderPadding: EdgeInsets.zero,
      expansionCallback: (index, v) {},
      children: buildChildren(),
    );
  }

  List<ExpansionPanel> buildChildren() {
    return FaqRepo().getFaqs().map((faqItem) {
      return ExpansionPanel(
          isExpanded: faqItem.isExpanded,
          headerBuilder: (context, b) {
            return buildHeader(faqItem);
          },
          body: Text(faqItem.expanded));
    }).toList();
  }

  Container buildHeader(FaqItem faqItem) {
    return Container(
      decoration: BoxDecoration(
          color: Colors.teal, borderRadius: BorderRadius.circular(16)),
      child: Center(
        child: Text(
          faqItem.header,
          style: TextStyle(color: Colors.white),
        ),
      ),
    );
  }
```

With the current implementation of ExpansionPanelList/ExpansionPanel, I can not achieve my goal.

**Comment by SpajicM** [comment]
What exactly would you like to customize, which parameters?

**Comment by sanisloandras** [comment]
I don't want the content of each item to be in a "card" with elevation.
I want the header to have a rounded corner, but with the "expand/collapse" icon part of that container.
It's pretty obvious if you look at the screenshots.

**Comment by iapicca** [comment]
Hi @sanisloandras 
This platform is not meant for assistance on personal code. Please see https://flutter.dev/community for resources and asking questions like this,
you may also get some help if you post it on Stack Overflow and if you need help with your code, please see https://www.reddit.com/r/flutterhelp/
Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by sanisloandras** [comment]
@iapicca 
I opened the issue under this category: 
"I want help writing my application.
You have a question for how to achieve a particular effect, or you need help with using a particular API."
And that is exactly what I need.

I already analyse the code of ExpansionPanelList and know the asnwer: it's not possible. Well, it is, if I copy paste the ExpansionPanelList, and modify it according to my needs.
But in my modest opinion, a class should be easily extendable without modifying the class itself.

So please, leave it, or move it the "future request" section, or whatever else.

**Comment by iapicca** [comment]
Hi @SpajicM 
I'm not sure that want you want to achieve isn't already possible with the current implementation,
could you please provide a runnable code sample (not a snippet of production code)
Thank you

**Comment by SpajicM** [comment]
@iapicca You've mentioned a wrong person there.

@sanisloandras I agree with you completely. I had to copy the ExpansionPanelList many times. I was just pointing out that a feature request phased as "i want this" won't get you an expected result - as seen above - and should be broken down into specific customization features and parameters..

I think what you would want ExpansionPanel to expose:
- card elevation (work in progress #51321)
- shape (OvalClip doesn't work)
- icon and icon color (without overriding theme data) - there have been two PRs, both abandoned https://github.com/flutter/flutter/pull/32887 https://github.com/flutter/flutter/pull/35416
- panel background color (without overriding theme data)

There's an ongoing discussion about ExpansionPanelList at https://github.com/flutter/flutter/issues/51321 where you could add your ideas.

**Comment by iapicca** [comment]
Hi @SpajicM 
my apologise for tagging you inadvertently 
and thank you for spotting the issue you mentioned in your last message 


Hi @sanisloandras 
please follow up on the [issue](https://github.com/flutter/flutter/issues/51321) referred [above](https://github.com/flutter/flutter/issues/60438#issuecomment-651388242)
Please follow up on that issue, I'm closing the current one as duplicate.
If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by Tokariev** [comment]
Hello @sanisloandras my solution for this problem looks like this -> First of all I copied the class ExpansionPanelList and make my own (CustomExpansionPanelList). Then I removed shadow from the custom class and change color (for grey). I didn't find how can I add a border radius but I was able to make the same thing with Container and Border radius inside of them. The result looks pretty well:
![Bildschirmfoto 2021-03-05 um 21 07 21](https://user-images.githubusercontent.com/26324819/110170878-1fe07880-7dfb-11eb-8328-76b4bb20a962.png)
